### ISSUE TITLE
Can not launch the target in new terminal 

### ISSUE THREAD

**Issue by tourkit** [microsoft/vscode-cmake-tools#3220]
### Brief Issue Summary

 I use Cmake Tools "launch" to build and open the program in one go. it compiles then open a "Cmake" terminal where it trys to launch the program .

Since IDK what happened, after building, the program doesnt run from within the Cmake/Launch integrated terminal where as if i open a new ntergrated terminal and try to launch the progran there it works. TBH it works sometimes and sometimes not, is inconsistent.

### CMake Tools Diagnostics

```shell
{
  "os": "win32",
  "vscodeVersion": "1.79.2",
  "cmtVersion": "1.14.33",
  "configurations": [
    {
      "folder": "c:\\msys64\\home\\MYUSER\\old\\MYPROGRAM",
      "cmakeVersion": "3.22.0",
      "configured": true,
      "generator": "Ninja",
      "usesPresets": false,
      "compilers": {
        "C": "C:/msys64/mingw64/bin/gcc.exe",
        "CXX": "C:/msys64/mingw64/bin/g++.exe"
      }
    }
  ],
  "cpptoolsIntegration": {
    "isReady": true,
    "hasCodeModel": true,
    "activeBuildType": "Debug",
    "buildTypesSeen": [
      "Debug"
    ],
    "requests": [
      "file:///c%3A/msys64/home/MYUSER/old/MYPROGRAM/main.cpp"
    ],
    "responses": [
      {
        "uri": "file:///c%3A/msys64/home/MYUSER/old/MYPROGRAM/main.cpp",
        "configuration": {
          "includePath": [
            "c:/msys64/home/MYUSER/old/MYPROGRAM",
            "c:/msys64/home/MYUSER/old/MYPROGRAM/src",
            "c:/msys64/home/MYUSER/old/MYPROGRAM/include",
            "c:/msys64/home/MYUSER/old/MYPROGRAM/include/vendor",
            "c:/msys64/home/MYUSER/old/MYPROGRAM/include/vendor/gl3w/include"
          ],
          "defines": [
            "GLFW_DLL"
          ],
          "compilerPath": "c:/msys64/mingw64/bin/g++.exe",
          "compilerArgs": [],
          "compilerFragments": [
            "-g",
            "-DIMGUI_IMPL_OPENGL_LOADER_GL3W  ",
            "-std=gnu++20"
          ]
        }
      }
    ],
    "partialMatches": [],
    "targetCount": 1,
    "executablesCount": 1,
    "librariesCount": 0,
    "targets": [
      {
        "name": "MYPROGRAM",
        "type": "EXECUTABLE"
      }
    ]
  },
  "settings": [
    {
      "communicationMode": "automatic",
      "useCMakePresets": "auto",
      "configureOnOpen": true
    }
  ]
}
```

### Debug Log

```shell
[main] Building folder: MYPROGRAM 
[main] Saving open files before configure/build
[build] Starting build
[driver] Start build MYPROGRAM
[proc] Executing command: C:\msys64\mingw64\bin\cmake.EXE --build c:/msys64/home/MYUSER/old/MYPROGRAM/build --config Debug --target MYPROGRAM --
[proc]   with environment: {"ALLUSERSPROFILE":"C:\\ProgramData","APPDATA":"C:\\Users\\MYUSER\\AppData\\Roaming","APPLICATION_INSIGHTS_NO_DIAGNOSTIC_CHANNEL":"1","COMPUTERNAME":"SYSTEM","ComSpec":"C:\\Windows\\system32\\cmd.exe","CommonProgramFiles":"C:\\Program Files\\Common Files","CommonProgramFiles(x86)":"C:\\Program Files (x86)\\Common Files","CommonProgramW6432":"C:\\Program Files\\Common Files","DriverData":"C:\\Windows\\System32\\Drivers\\DriverData","ELECTRON_RUN_AS_NODE":"1","HOMEDRIVE":"C:","HOMEPATH":"\\Users\\MYUSER","LOCALAPPDATA":"C:\\Users\\MYUSER\\AppData\\Local","LOGONSERVER":"\\\\SYSTEM","NUMBER_OF_PROCESSORS":"12","ORIGINAL_XDG_CURRENT_DESKTOP":"undefined","OS":"Windows_NT","PATHEXT":".COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC","PROCESSOR_ARCHITECTURE":"AMD64","PROCESSOR_IDENTIFIER":"Intel64 Family 6 Model 158 Stepping 10, GenuineIntel","PROCESSOR_LEVEL":"6","PROCESSOR_REVISION":"9e0a","PSModulePath":"C:\\Program Files\\WindowsPowerShell\\Modules;C:\\Windows\\system32\\WindowsPowerShell\\v1.0\\Modules","PUBLIC":"C:\\Users\\Public","Path":"C:\\Windows\\system32;C:\\Windows;C:\\Windows\\System32\\Wbem;C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\;C:\\Windows\\System32\\OpenSSH\\;C:\\Program Files\\Git\\cmd;C:\\Users\\MYUSER\\AppData\\Local\\Microsoft\\WindowsApps;C:\\Program Files\\Neovim\\bin;C:\\msys64\\mingw64\\bin;C:\\msys64\\mingw64\\bin","ProgramData":"C:\\ProgramData","ProgramFiles":"C:\\Program Files","ProgramFiles(x86)":"C:\\Program Files (x86)","ProgramW6432":"C:\\Program Files","SESSIONNAME":"Console","SystemDrive":"C:","SystemRoot":"C:\\Windows","TEMP":"C:\\Users\\MYUSER\\AppData\\Local\\Temp","TMP":"C:\\Users\\MYUSER\\AppData\\Local\\Temp","USERDOMAIN":"SYSTEM","USERDOMAIN_ROAMINGPROFILE":"SYSTEM","USERNAME":"MYUSER","USERPROFILE":"C:\\Users\\MYUSER","VSCODE_AMD_ENTRYPOINT":"vs/workbench/api/node/extensionHostProcess","VSCODE_CODE_CACHE_PATH":"C:\\Users\\MYUSER\\AppData\\Roaming\\Code\\CachedData\\695af097c7bd098fbf017ce3ac85e09bbc5dda06","VSCODE_CRASH_REPORTER_PROCESS_TYPE":"extensionHost","VSCODE_CWD":"C:\\Users\\MYUSER\\AppData\\Local\\Programs\\Microsoft VS Code","VSCODE_HANDLES_UNCAUGHT_ERRORS":"true","VSCODE_IPC_HOOK":"\\\\.\\pipe\\7efa7c941ce40a3dcf2eecb791ae2089-1.79.2-main-sock","VSCODE_L10N_BUNDLE_LOCATION":"","VSCODE_NLS_CONFIG":"{\"locale\":\"en-us\",\"osLocale\":\"fr-fr\",\"availableLanguages\":{},\"_languagePackSupport\":true}","VSCODE_PID":"6472","windir":"C:\\Windows","CMT_MINGW_PATH":"C:\\msys64\\mingw64\\bin","NINJA_STATUS":"[%s/%t %p :: %e] "}
[build] ninja: no work to do.
[driver] Build completed: 00:00:00.070
[cmakefileapi-parser] Read reply folder: c:\msys64\home\MYUSER\old\MYPROGRAM\build\.cmake\api\v1\reply
[cmakefileapi-parser] Found index files: ["cache-v2-960580d75fa978582159.json","cmakeFiles-v1-9d863760ca36b49dd54e.json","codemodel-v2-66ff26d0c044c5177e12.json","directory-.-Debug-d0094a50bb2071803777.json","index-2023-06-26T20-49-47-0829.json","target-MYPROGRAM-Debug-7bcb04367fe9699fb316.json","toolchains-v1-99b97b9bc1254da480bd.json"]
[build] Build finished with exit code 0
[cache] Reading CMake cache file c:/msys64/home/MYUSER/old/MYPROGRAM/build/CMakeCache.txt
[cache] Parsing CMake cache string
[extension] [7677] cmake.build finished (returned 0)
```

### Additional Information

![image](https://github.com/microsoft/vscode-cmake-tools/assets/63537158/fd625081-8b35-46e9-8195-5488c875837a)
![image](https://github.com/microsoft/vscode-cmake-tools/assets/63537158/cd9fa5fd-c5d0-4878-8537-22423375572d)

**Comment by benmcmorran** [comment]
Thanks for following up. We'll investigate and let you know if we need any additional details.

**Comment by yann-qu** [comment]
I have a similar problem. If I launch the program using the bottom launch button (the small triangle symbol), the program will exit without any output. But if I launch the program in the terminal created by myself, the program will run properly.

The difference of the two terminal is environment variable, which could be displayed by command `echo $env:path`. In the terminal created by the plugin, which named like "CMake/Launch - xxx", the environment variable is strange, it has two mingw and the sequence is wrong.

I have to switch to version `v1.13.45` to avoid the problem.

**Comment by Lan-Friend** [comment]
@yann-qu 
I am also facing the same issue. In fact, I have been trying for a whole day. At the first, I thought it was due to my actions causing a discrepancy between the environment variables in the terminal opened when clicking the triangle and the environment variables in a newly opened terminal. However, after rolling back to the previous version(`v1.13.45`), I found that the environment variables are consistent. I believe this is a bug in the new version.

**Comment by tourkit** [comment]
for ppl with similar pb I meanwhile found an alternative solution in reddit see here https://www.reddit.com/r/vscode/comments/14k6d1w/comment/jppu4zb/?context=3

thx for following up @benmcmorran

**Comment by xuehao** [comment]
On Windows 11, I create a new windows account and the issue disappears. But after using some days, the issue happens again. So I think this may be any config or cache issue.

On Ubuntu/macOS, I never encounter this kind of issues.

**Comment by gcampbell-msft** [comment]
Related to #3192

**Comment by tourkit** [comment]
@gcampbell-msft : hey, switched to pre release, SHIFT+F5 (CMake launch) , same result, got my head busy elsewhere nowadays cant remember much anything else helpfull to check sorry.

![image](https://github.com/microsoft/vscode-cmake-tools/assets/63537158/7cd40db9-f3e9-4e8c-a53f-9f05327f51e4)

**Comment by snehara99** [comment]
> it has two mingw and the sequence is wrong.

Hi @yann-qu . Would you be able to share that path that seemed incorrect?

**Comment by snehara99** [comment]
@tourkit @yann-qu @Lan-Friend @xuehao Could one or more of you provide detailed repro steps from the beginning. Starting from what tools you installed, where  they are installed, etc. to a small sample project that can repro the issue? This issue is difficult to investigate without being able to reproduce the issue.

**Comment by Lan-Friend** [comment]
@snehara99 
Em...，I'm not very good at describing problems, so I will now directly reproduce it:

My problem is that there is an error reading system environment variables in the new version.

CMakeLists.txt
```cmake
# other code
    set(myPath $ENV{PATH})
    STRING(REGEX REPLACE "[A-Z]:\\\\" "\\n" resPath ${myPath})
    message(STATUS ${resPath})
```

If executed using CMake Tools Version 1.15.31, the following content will look like:

```output
// Other identical paths
IDE\USE\NodeJs
IDE\USE\cmake-3.26.4-windows-x86_64\bin
Text\College\Program_Design\Qt\6.2.4\msvc2019_64\bin
IDE\USE\Git\cmd
IDE\USE\LLVM\bin
IDE\USE\mingw64 x86_64-13.1.0-release-posix-seh-msvcrt-rt_v11-rev1\bin
IDE\USE\mingw64 x86_64-13.1.0-release-posix-seh-msvcrt-rt_v11-rev1\bin
```

This is not the correct 'PATH'

Then when I roll back to Version 1.13.45, the output will look like this:
```output
// Other identical paths
IDE\USE\mingw64 x86_64-13.1.0-release-posix-seh-msvcrt-rt_v11-rev1\bin
IDE\USE\NodeJs
IDE\USE\cmake-3.26.4-windows-x86_64\bin
Text\College\Program_Design\Qt\6.2.4\mingw_64\bin
Text\College\Program_Design\Qt\6.2.4\msvc2019_64\bin
IDE\USE\OpenCV_4.8.0\x64\mingw\bin
IDE\USE\Git\cmd
IDE\USE\LLVM\bin
```

This is the correct 'PATH'

In both output results, I have removed most of the identical paths.

**Comment by snehara99** [comment]
@Lan-Friend thank you for this info, it'll help guide my investigation.

**Comment by xuehao** [comment]
Hi,

This is my vscode version:
- `VS Code v1.83.1`
- `CMake Tools v1.16.13`

I'm using the newest [MSYS2 UCRT toolchain](https://packages.msys2.org/groups/mingw-w64-ucrt-x86_64-toolchain).
![Screenshot 2023-10-13 094920](https://github.com/microsoft/vscode-cmake-tools/assets/2335731/29a514f0-d2a2-4acf-9953-6682595f011e)

This is my demo project with this kind of issue: [2.2_Add2.zip](https://github.com/microsoft/vscode-cmake-tools/files/12888684/2.2_Add2.zip)

![Screenshot 2023-10-13 094746](https://github.com/microsoft/vscode-cmake-tools/assets/2335731/3da54f7c-2667-4aef-9fed-1a1b339b263a)

**Comment by snehara99** [comment]
Hi @xuehao thank you for the sample project!

**Comment by snehara99** [comment]
Hello @xuehao @yann-qu @Lan-Friend @ebkgne, could you please install the new pre-release version of the CMakeTools extension and see if your issue has been resolved?

**Comment by Lan-Friend** [comment]
@snehara99 
I switched to the pre-release version 1.16.14 of CMakeTools and conducted testing.
If we only consider the integrity of 'PATH', there is basically no problem.

CMakeLists.txt
```CMake
# other code
    set(myPath $ENV{PATH})
    STRING(REGEX REPLACE "[A-Z]:\\\\" "\\n" resPath ${myPath})
    message(STATUS ${resPath})
```
I expect it to output as follows：

```expected
# other paths
IDE\USE\mingw64 x86_64-13.1.0-release-posix-seh-msvcrt-rt_v11-rev1\bin
Program Files\Azure Data Studio\bin
Program Files\dotnet
IDE\USE\NodeJs
IDE\USE\cmake-3.26.4-windows-x86_64\bin
```

But it repeats some of the content

```output
# other paths
IDE\USE\mingw64 x86_64-13.1.0-release-posix-seh-msvcrt-rt_v11-rev1\bin
IDE\USE\mingw64 x86_64-13.1.0-release-posix-seh-msvcrt-rt_v11-rev1\bin
IDE\USE\mingw64 x86_64-13.1.0-release-posix-seh-msvcrt-rt_v11-rev1\bin
Program Files\Azure Data Studio\bin
Program Files\dotnet
IDE\USE\NodeJs
IDE\USE\cmake-3.26.4-windows-x86_64\bin
Program Files\Azure Data Studio\bin
```

**Comment by xuehao** [comment]
Sorry, the pre-release version with the same issue.

![Screenshot 2023-10-18 141355](https://github.com/microsoft/vscode-cmake-tools/assets/2335731/8c90c204-adc9-4ebd-a1a4-25b32956e6ee)

**Comment by snehara99** [comment]
@Lan-Friend is your launch issue resolved?

**Comment by Lan-Friend** [comment]
@snehara99 It can be considered resolved since the environment variable ’PATH‘ is fully preserved. However, the content of the 'PATH' may be duplicated.

**Comment by snehara99** [comment]
@xuehao The project you shared has too many dependencies and introduces a lot of variables. Could you share a project that doesn't require these dependencies? Additionally, could you share which previous version of CMakeTools does not have this issue? 

I was able to get your project working using WSL, btw.

**Comment by xuehao** [comment]
Hello @snehara99,

I reproduced this issue. My project depends on SDL2. When I remove SDL dependency, the project works ok.

![without-sdl2](https://github.com/microsoft/vscode-cmake-tools/assets/2335731/27d0831e-f32c-4d2c-9876-5cadaab9a38f)

When I add SDL again, the project fails again.

![with-sdl2](https://github.com/microsoft/vscode-cmake-tools/assets/2335731/e332624d-f1a2-4ca1-97e0-ff34edcfcd15)

I don't know the reason. Maybe it's not the extension problem. Is it possible the PkgConfig issue?

You can try the minimal sample project. Comment/Uncomment the SDL config: 
[Add2.zip](https://github.com/microsoft/vscode-cmake-tools/files/13035592/Add2.zip)

By the way, to install SDL in MSYS2 UCRT:
```
pacman -Sy mingw-w64-ucrt-x86_64-toolchain     \
           mingw-w64-ucrt-x86_64-SDL2          \
           mingw-w64-ucrt-x86_64-SDL2_image    \
           mingw-w64-ucrt-x86_64-SDL2_mixer    \
           mingw-w64-ucrt-x86_64-SDL2_ttf
```

**Comment by snehara99** [comment]
I will now close this issue because we cannot confirm if the original problem is fixed or not. @ebkgne please open a new issue if you still need this to be investigated.

**Comment by qiuliw** [comment]
我遇到了同样的问题

**Comment by yanghhhhhhh** [comment]
Hi @qiuliw , thank you for your comment. We are investigating this issue. Please follow up on this new issue https://github.com/microsoft/vscode-cmake-tools/issues/4542. Thank you.
